### PR TITLE
[TT-207] - Generate Report Functionality for Admins

### PR DESCRIPTION
**Pull request by blackinferno1000**
[Ticket](https://ritservices.atlassian.net/browse/TT-207)

Functionality for exporting a committee's charges and minutes to CSV was needed.
I found an implementation to do it on the client side as nom packages can't use fs on browsers.
Some tests might be needed if more complex data is added but it should be fine for now.

### CODE REVIEW

File: src/assets/charges.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+NAME,DESCRIPTION,CREATED_AT,ASSIGNED_TO,STATUS
+test,,,,
```
Comment (lukasyelle) [inline src/assets/charges.csv]: Are these files necessary?

File: src/pages/Admin.vue [hunk 2]
```diff
@@ -299,6 +302,91 @@ export default {
...
+      })
+
+      var header = {name: 'NAME', description: 'DESCRIPTION', timestamp: 'CREATED_AT', assignment: 'ASSIGNED_TO', status: 'STATUS'}
+      var header2 = {title: 'TITLE', relatedCharges: 'RELATED CHARGES', name: 'COMMITTEE NAME', timestamp: 'CREATED_AT'}
+
+      var itemsFormatted = []
+      var itemsFormatted2 = []
+
+      // format the data
+      this.committeeCharges.forEach((charge) => {
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: When running this locally I get a console error saying that `committeeCharges` is null the first time I click on the Get CSV button. This might be because the request on line 353 doesn't complete before this line is reached. 

On Safari I'm never able to get the CSV for charges, but it works on Firefox and Chrome.

File: src/pages/Admin.vue [hunk 3]
```diff
@@ -299,6 +302,91 @@ export default {
...
+      if (headers) {
+        items.unshift(headers)
+      }
+
+      // Convert Object to JSON
+      var jsonObject = JSON.stringify(items)
+
+      var csv = this.convertToCSV(jsonObject)
+
+      var exportedFilenmae = fileTitle + '.csv' || 'export.csv'
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: typo in variable name

File: src/pages/Admin.vue [hunk 4]
```diff
@@ -299,6 +302,91 @@ export default {
...
+      })
+      this.committeeMinutes.forEach((minute) => {
+        var chargesIds = 'N/A'
+        if (minute.charges.length > 0) {
+          chargesIds = ''
+          minute.charges.forEach((charge) => {
+            chargesIds += `${charge.id} `
+          })
+        }
+        itemsFormatted2.push({title: minute.title, relatedCharges: chargesIds, name: minute.committee_id, timestamp: minute.date})
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: `minute.date` is formatted as a number so it needs to be converted to a date.

File: src/pages/Admin.vue [hunk 5]
```diff
@@ -438,6 +506,30 @@ export default {
     }
   },
   sockets: {
+    get_charges: function (data) {
+      this.committeeCharges = data
+      this.committeeCharges.forEach((charge) => {
+        this.itemsFormatted.push({name: charge.title, description: charge.description, timestamp: charge.created_at, assignment: charge.committee, status: charge.status})
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: `itemsFormatted` gets appended to whenever charges are received from the backend. Because of this if you click "Get CSV" more than once on this page you will have duplicates. Is it necessary to have the `itemsFormatted` field?

File: src/pages/Admin.vue [hunk 6]
```diff
@@ -438,6 +506,30 @@ export default {
...
+            chargesIds += `${charge.id} `
+          })
+        }
+        let date = new Date(parseInt(minute.date))
+        let fdate = (date.getMonth() + 1) + '/' + date.getDate() + '/' + date.getFullYear()
+        this.itemsFormatted2.push({title: minute.title, relatedCharges: chargesIds, name: minute.committee_id, timestamp: fdate})
+      })
+
+      this.exportCSVFile(this.header, this.itemsFormatted, this.fileTitle) // call the exportCSVFile() function to process the JSON and trigger the download
+      this.exportCSVFile(this.header2, this.itemsFormatted2, this.fileTitle2) // call the exportCSVFile() function to process the JSON and trigger the download
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: Here `exportCSVFile` is getting called for both the minutes and the charges. This could result in behavior where the `get_minutes` request completes before the `get_charges` request does, and prints an empty spreadsheet or a spreadsheet with the charges from a different committee.

File: src/pages/Admin.vue [hunk 7]
```diff
@@ -438,6 +506,30 @@ export default {
...
+        var chargesIds = 'N/A'
+        if (minute.charges.length > 0) {
+          chargesIds = ''
+          minute.charges.forEach((charge) => {
+            chargesIds += `${charge.id} `
+          })
+        }
+        let date = new Date(parseInt(minute.date))
+        let fdate = (date.getMonth() + 1) + '/' + date.getDate() + '/' + date.getFullYear()
+        this.itemsFormatted2.push({title: minute.title, relatedCharges: chargesIds, name: minute.committee_id, timestamp: fdate})
```
Comment (MontgomeryWatts) [inline src/pages/Admin.vue]: Same concern as with the `itemsFormatted` field.
